HealthcareSystem/GraphicalEditorServer/Controllers/EquipmentController.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Backend.Service.RoomAndEquipment;
+using GraphicalEditorServer.DTO;
+using Newtonsoft.Json;
+using Model.Manager;
+
+
+namespace GraphicalEditorServer.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class EquipmentController : ControllerBase
+    {
+        private readonly INonConsumableEquipmentService _nonConsumableEquipmentService;
+        private readonly IConsumableEquipmentService _consumableEquipmentService;
+
+        public EquipmentController(INonConsumableEquipmentService nonConsumableEquipmentService, IConsumableEquipmentService consumableEquipmentService)
+        {
+            _nonConsumableEquipmentService = nonConsumableEquipmentService;
+            _consumableEquipmentService = consumableEquipmentService; 
+        }
+
+        [HttpPost]
+        public ActionResult AddEquipment([FromBody] String test)
+        {
+            string[] atributes = test.Split(",");
+            String final = "{";
+            final += test;
+            final += "}";
+
+            if (atributes.Length<3) {
+                 NonConsumableEquipment nonCosumableEquipment = JsonConvert.DeserializeObject<NonConsumableEquipment>(final);
+
+                _nonConsumableEquipmentService.newNonConsumableEquipment(nonCosumableEquipment);
+            }
+            else
+            {
+                ConsumableEquipment cosumableEquipment = JsonConvert.DeserializeObject<ConsumableEquipment>(final);
+
+                _consumableEquipmentService.newConsumableEquipment(cosumableEquipment);
+            }
+            return Ok();
+        }
```

Comment:
What is going on here? Are you having issues with deserialization?

Reply:
- The goal was to establish a connection with our WPF and database
- The problem with different classes in Backend is that we were bounded with the existing SiMS project which we did not want to change in order not to break something that is used by the web team. I agree, it can make things more complicated.
- As serialization/deserialization is concerned, we were having some weird problems with it, so this is the only way we made it to work

Reply:
What sort of problems? Maybe we can help, this is really odd and shouldn't be happening. 
Also, I don't think Web is using equipment at all right now, you can probably do whatever you want there. Although I feel like  merging the initial project with the established connection to develop and then changing it on a different branch might be a better approach right now.

Reply:
We tried serialization of the object we want to send and deserialize in controllers, but it was constantly giving us exceptions. Maybe, it is worth trying again. Since that approach did not work we converted an object into json string, but these brackets {} could not be added, so we added them into the controller with [FromBody] annotation and did deserialization. It was pretty weird, so I believe it is worth trying again.
Yes, we are using the Model classes for now. Adding DTOs is in the plan, but we are waiting to see what will the next require from us, so if needed, refactoring will be done.
We will need to consult with Web before we start making any changes. I will tell our SM to investigate that part on the next meeting.

Reply:
It's the hierarchy probably, the serializer can't tell what the actual type of the object is. Having DTOs without the hierarchy (or maybe even removing it from the model, I don't think it actually adds much) is probably the way to go in the future.

Reply:
I am planning to give a new try soon, so I will see if there something I missed. I don't really think it is the hierarchy, I have already done serialization of objects in the hierarchy and it worked fine.

Reply:
No, I'm pretty sure it is the hierarchy. Look at what this method is doing - checking the length of the message and then instantiating different types based on that. I assume that the type of the parameter that couldn't get deserialized was the base of the hierarchy? JSON wouldn't know what type to instantiate then, it can't really deal with polymorphism out of the box.
Splitting this method into two might make these issues go away though, and it would make sense anyway.

Reply:
If we are talking only about this method and checking the number of parameters, that is correct. That is the reason why it was made like that in the first place. I will split this method into 2. Since adding new items is not required from us ( for now, and there is a chance it will stay like that in the future, but I cannot tell for sure until new sprints come) I believe investing any more time on furthermore refactoring of add functionality would be a waste of time since we are limited with the deadline. After we consult with the Web team, we will maybe do redesign of the model and addition of DTO, so that serialization/deserialization becomes smoother.

Reply:
But I see the methods are still receiving strings, it still won't deserialize these? That's really odd then. 
But yeah, I agree with not refactoring anything here, and probably not wasting any more time on whatever is wrong with the serialization either. I'm fine with merging this branch whenever the conflict is solved.

Reply:
I've tried a few different options for transferring data, but this is the only one that seems to work. I am open to refactoring as soon as I see a chance to do it smoothly.
Could you take a look at the message I left on Discord regarding the conflict?